### PR TITLE
Add documentation pages for KResponsiveWindow and KResponsiveElement

### PR DESCRIPTION
**Pull request by MisRob**
## Description

Updates library documentation:

- Updates _Layouts_ page: Some sections related to  `KResponsiveWindowMixin` were moved to its new page, and _Layouts_ page now only contains general breakpoints information with newly added references to the two new pages. 
- Adds two new pages:
  - _KResponsiveWindow_ page: To a large extent, existing documentation from the _Layouts_ page was used.
  - _KResponsiveElement_ page: I wrote down some basic information and usage but didn't create an example since it's quite simple, and also, currently, we rather discourage using it.

Two new pages were prepared with keeping https://github.com/learningequality/kolibri-design-system/issues/288 (allow mixins logic to be used via composition API) in mind. Therefore, pages are called _KResponsiveWindow_/_KResponsiveElement_ and not `KResponsiveWindowMixin`/`KResponsiveElementMixin` since in the near future we should be able to access this logic in an alternative way to mixins, via composables (`useKResponsiveWindow`, `useKResponsiveElement`). Relatedly, all information provided on new pages is presented in a way applicable to both existing mixins as well as to new composables to be implemented. Usage details specific to mixins can be found in the _Usage -> As mixins_ section of pages. This prepares the ground for a new sub-section, _Usage-> As composables_ to be added as part of https://github.com/learningequality/kolibri-design-system/issues/288.

#### Issue addressed

I mentioned performance troubles on _KResponsiveElement_ page so this also closes https://github.com/learningequality/kolibri-design-system/issues/237

## Steps to test

Read
- [ ] _Layout: Responsiveness_: https://deploy-preview-362--kolibri-design-system.netlify.app/layout/#responsiveness
- [ ] _KResponsiveElement_: https://deploy-preview-362--kolibri-design-system.netlify.app/kresponsiveelement/
- [ ] _KResponsiveWindow_: https://deploy-preview-362--kolibri-design-system.netlify.app/kresponsivewindow/

Is it easy to follow? Can you spot any typos, recommend grammar improvements?

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical and brittle code paths are covered by unit tests
- [x] The change has been added to the `changelog`

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] Are there tests for this change?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?
- [ ] _Add other things to check for here_

## Post-merge updates and tracking

- [ ] Learning Platform update PR is submitted
- [ ] Learning Platform update PR is merged
- [ ] Studio update PR is submitted
- [ ] Studio update PR is merged
- [ ] Data Portal update PR is submitted
- [ ] Data Portal update PR is merged

### CODE REVIEW

**Comment by MisRob** [comment]
I'm targeting the release branch because I think it may be useful to have these updates live. After we merge this, I will merge the release branch into the develop branch so it's also available when @akolson starts working on https://github.com/learningequality/kolibri-design-system/issues/288 as that work will target the develop branch.

**Comment by radinamatic** [comment]
Nice worthy additions, docs build well locally, and I did not notice any overt grammar errors:

KResponsive
 
Given that the KDS documentation is a 👶🏽 of many with varying _parenting_ (writing 🙂) styles that we never had bandwidth to unify from the technical writing point of view, I'm vary of making styling and editing suggestions for these specific components as those will probably be valid for all the rest.

**Review by radinamatic — APPROVED** [review]
LGTM @MisRob, thank you!

**Review by nucleogenesis — APPROVED** [review]
This is excellent @MisRob  - I really love the "change your browser size to see this in action" example you gave :)

**Comment by MisRob** [comment]
@nucleogenesis I just moved that example from the Layouts page, and I suspect that @indirectlylit is its author :)

File: docs/pages/kresponsivewindow.vue [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+      <!-- eslint-disable -->
+      <!-- prevent prettier from changing indentation -->
+      <DocsShowCode language="javascript">
+        import KResponsiveWindowMixin from 'kolibri-design-system/lib/KResponsiveWindowMixin';
+
+        export default {
+          mixins: [KResponsiveWindowMixin]
+        };
+      </DocsShowCode>
```
Comment (indirectlylit) [inline docs/pages/kresponsivewindow.vue:39]: the scrollbar/layout issue below is addressed here:

https://github.com/learningequality/kolibri-design-system/pull/358/files#diff-68f592b61f5d68cabe12194516b8ac51b6f9aa358d2a08961b5d0a698210c9ea

image

File: docs/pages/kresponsivewindow.vue [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      <!-- eslint-disable -->
+        <!-- prevent prettier from changing indentation -->
+        <DocsShowCode language="html">
+          &lt;div class="box" :style="boxStyle"&gt;
+            Box 1
+          &lt;/div&gt;
+          &lt;div class="box" :style="boxStyle"&gt;
+            Box 2
+          &lt;/div&gt;
+        </DocsShowCode>
```
Comment (indirectlylit) [inline docs/pages/kresponsivewindow.vue:60]: this block was updated here:

https://github.com/learningequality/kolibri-design-system/pull/358/files#diff-1987289818bbdf91ad0f1113461d08d23be49a2a302a2df922b2e79a0a5f785d

**Review by indirectlylit — COMMENTED** [review]
looks great!

just a heads up that https://github.com/learningequality/kolibri-design-system/pull/358 might start getting a bit stale as the docs evolve

**Comment by MisRob** [comment]
Ah I'm sorry @indirectlylit, I knew about that PR. I will assign myself and will do my best to review soon and resolve conflicts.

**Comment by MisRob** [comment]
Thanks for pointing out those places

**Comment by indirectlylit** [comment]
no worries! just wanted to give a heads up - I know there are always a lot of things happening

**Pull request merged**